R/yahoo_franchises.R

```diff
@@ -0,0 +1,62 @@
+#' Get a dataframe of franchise data
+#'
+#' @param conn a conn object created by `ff_connect()`
+#'
+#' @examples
+#' \donttest{
+#' try({ # try only shown here because sometimes CRAN checks are weird
+#'   yahoo_conn <- ff_connect(platform = "yahoo", league_id = "77275", token = NULL)
+#'   ff_franchises(yahoo_conn)
+#' }) # end try
+#' }
+#' @describeIn ff_franchises Yahoo: Returns franchise data.
+#' @export
+ff_franchises.yahoo_conn <- function(conn) {
+  glue::glue("leagues;league_keys={conn$league_key}/teams") %>%
+    yahoo_getendpoint(conn) %>%
+    {
+      .yahoo_process_franchises_response(.$xml_doc)
+    }
+}
+
+.yahoo_process_franchises_response <- function(xml_doc) {
```

Comment:
Token you gave me expired before I could fully flesh out this example, but here's an alternative way to approach this problem - convert xml to list and use tidyr/dplyr/purrr to unnest out the dataframe you want

```r
  x <- xml_doc %>%
    xml2::as_list() %>%
    purrr::pluck("fantasy_content", "leagues","league","teams") %>%
    tibble::tibble() %>%
    tidyr::unnest_wider(1) %>%
    dplyr::select(
      franchise_id = team_id,
      franchise_name = name,
      division_id = division_id,
      managers
    ) %>%
    tidyr::unnest_longer(c(franchise_id, franchise_name, division_id)) %>%
    dplyr::mutate(
      managers = purrr::map(managers, ~ tibble::tibble(.x) %>% tidyr::unnest_wider(1))
    ) %>%
    tidyr::hoist(
      managers,
      "user_name" = "nickname",
      "user_id" = "guid",
      "user_felo" = "felo_score"
    ) %>%
    tidyr::unnest_longer(c(user_name, user_id, user_felo)) %>%
    tidyr::unnest_longer(c(user_name, user_id, user_felo))
```

Reply:
I spent a few hours reading up on the tidyverse but I just can't for the life of me figure out how to work with the as_list xml output.  I'm going to go with the xml_find_all solutions for now since they're super intuitive for the uninitated.

Reply:
the un-tidy-doctrinated :joy: that's fine. Let's put in getElement in all the relevant places and call it the finish line for this PR?

Reply:
Maybe useful for future reading if you want to look at these: https://r4ds.hadley.nz/rectangling